packages/go_router/lib/src/router.dart

```diff
@@ -507,8 +481,9 @@ class GoRouter extends ChangeNotifier implements RouterConfig<RouteMatchList> {
   ///
   /// This method throws when it is called during redirects.
   static GoRouter of(BuildContext context) {
-    final InheritedGoRouter? inherited =
-        context.dependOnInheritedWidgetOfExactType<InheritedGoRouter>();
+    final InheritedGoRouter? inherited = context
+        .getElementForInheritedWidgetOfExactType<InheritedGoRouter>()
+        ?.widget as InheritedGoRouter?;
```

Comment:
```suggestion
  final InheritedGoRouter? inherited = maybeOf(context)
```

Reply:
Oh... I missed the semicolon ... sorry about that.

Reply:
ha, it's ok, presubmit got our backs

Reply:
oops it looks like cli is failing, I would have to squash

Reply:
The missing comma will not solve it, because of missing cast.
I think we need to make the change like this to solve the code duplication.

```
  /// Find the current GoRouter in the widget tree.
  ///
  /// This method throws when it is called during redirects.
  static GoRouter of(BuildContext context) {
    final InheritedGoRouter? inherited = _getInheritedGoRouter(context);
    assert(inherited != null, 'No GoRouter found in context');
    return inherited!.goRouter;
  }

  /// The current GoRouter in the widget tree, if any.
  ///
  /// This method returns null when it is called during redirects.
  static GoRouter? maybeOf(BuildContext context) {
    final InheritedGoRouter? inherited = _getInheritedGoRouter(context);
    return inherited?.goRouter;
  }

  static InheritedGoRouter? _getInheritedGoRouter(BuildContext context) {
    return context
        .getElementForInheritedWidgetOfExactType<InheritedGoRouter>()
        ?.widget as InheritedGoRouter?;
  }
```
Sorry for the inconvenience, tried to help and made it much worse...

Reply:
I updated the code, don't worry, it just me been lazy to want to completely rely on github ui. haha

Reply:
Ok, your solution is much better.